managed-pointer/cpp-20/D0567R1.md

```diff
@@ -5,222 +5,384 @@
 
-The third is by passing a ``managed_ptr`` to an executor control structure such as ``async()``, as this will implicitly trigger a call to the ``put()`` member function on the **executor**. If the host CPU is not accessible at this point then this will implicitly trigger a call to``get()`` on the ``managed_ptr`` which will subsequently implicitly trigger a call to ``get()`` on the currently accessible **execution context**. This method is useful as it allows you to simply pass a  ``managed_ptr`` directly to a control structure without requiring an explicit call to ``put()``. This is implicit.
+The associated *execution context* and *executor* of a `managed_ptr` can also be accessible, the `managed_ptr` has a member function for querying wether the associated *execution context* and *executor* is accessible; `is_accessible()`.
 
-## Examples
+The data at the memory region of the `managed_ptr`s associated *execution context* and *executor* can be read and modified (if the type `T` is a non-const type) via the de-reference and pointer operators.
 
-There is a wide range of user cases that affect the design of a potential interface when it comes to asynchronous tasks and the movement of the data that those tasks require, however for the purposes of this paper the following example should provide a base requirement for the features this paper proposes.
+#### 3.3.4. Resource aquisition
 
-The following examples show how the features that will be presented in this paper can be used to move data from the host CPU memory region to the memory region of a GPU in order to perform an operation on said data via the **executor** interface.
+The `managed_ptr` must allocate memory within the memory region of any execution context which that `managed_ptr` is accessible on via that execution context's allocator. This does not have to be in tamdem or on construction of the `managed_ptr` instance, the point at which allocation happens for each execution context is implementation defined, the only requirement is that the memory must be allocated (if not already so) prior to the `managed_ptr` being accessed on said execution context. Additionally the `managed_ptr` must deallocate in all memory regions that the `managed_ptr` instance has been allocated on, during its destructor.
 
-The first example shows how this would look using the explicit interface for synchronisation operations (figure 3) and the second shows how this would look using the implicit interface for synchronisation operations (Figure 4).
+If a user wishes to explicitly allocate or deallocate memory for a `managed_ptr` instance on a particular execution context, this can be done by calling the allocation operations `allocate`, `then_allocate`, `deallocate` and `then_deallocate`. Each of which is an asycnhronous operation which performs an allocation or dallocation on a particular execution context and returns a future to the complletion of the operation.
```

Comment:
wishes -> wants
asycnhronous -> asynchronous
Each of which -> "Each function"
complletion -> "completion"